src/array.ts

```diff
@@ -1,31 +1,27 @@
-export const objectListForAttribute = <T>(
+export const groupByKey = <T>(
   objects: T[],
-  attribute: string
+  key: string
 ): Record<string, T[]> => {
-  return objects.reduce(
-    (objectListForAttribute: Record<string, T[]>, object: T) => {
-      objectListForAttribute[object[attribute]]
-        ? objectListForAttribute[object[attribute]].push(object)
-        : (objectListForAttribute[object[attribute]] = [object]);
-      return objectListForAttribute;
-    },
-    {}
-  );
+  return objects.reduce((grouped: Record<string, T[]>, object: T) => {
+    const keyValue = object[key];
+    grouped[keyValue] = grouped[keyValue]
```

Comment:
I tend to use functions that return values so I prefer the [spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) to the [Array.push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) method. Similarly I prefer `slice` to `splice` I think it leads to more functional coding which is more readable and less buggy.

I don't like long method chains either. I usually try to break up method chains beyond 3-5 depending on the complexity to another function.

Reply:
Ok that is a good rule to go by. I also like the spread - its so natural for me to default to any collections built in methods from Java... on a similar vein, I was having a lot of trouble using `string[]` over `Array<string>`, it seems the syntactical sugar generally improves readability. Also like the rule of having a length to start evaluating a refactor